File: base/src/main/java/com/ssz/baselibrary/model/local/DbHelper.java
package com.ssz.baselibrary.model.local;

import android.database.sqlite.SQLiteDatabase;

import com.ssz.baselibrary.app.helper.AppHelper;
import com.ssz.baselibrary.model.local.dao.DaoMaster;
import com.ssz.baselibrary.model.local.dao.DaoSession;

public final class DbHelper {

    private static volatile DbHelper instance;
    private static final String DB_NAME = "ssz_db_note";

    private SQLiteDatabase mDb;
    private DaoMaster mDaoMaster;
    private DaoSession mSession;

    public static DbHelper getInstance() {
        if (null == instance){
            synchronized (DbHelper.class){
                if (null == instance){
                    instance = new DbHelper();
                }
            }
        }
        return instance;
    }

    private DbHelper(){
       DbOpenHelper openHelper = new DbOpenHelper(AppHelper.getApplication(),DB_NAME,null);
       mDb = openHelper.getWritableDatabase();
       mDaoMaster = new DaoMaster(mDb);
       mSession = mDaoMaster.newSession();
    }


    public DaoSession getNewSession(){
        return mDaoMaster.newSession();
    }
}

File: app/src/main/java/com/ssz/studydemo/model/local/DbHelper.kt
package com.ssz.studydemo.model.local

import android.database.sqlite.SQLiteDatabase
import com.ssz.studydemo.app.AppContext
import com.ssz.studydemo.model.local.dao.DaoMaster
import com.ssz.studydemo.model.local.dao.DaoSession


/**
 * @author : zsp
 * time : 2019 09 2019/9/17 14:28
 */
class DbHelper private constructor(){

    private val mDb: SQLiteDatabase
    private val mDaoMaster: DaoMaster
    private val mSession: DaoSession

    companion object {
        private const val DB_NAME = "ssz_note"
        val INSTANCE : DbHelper by lazy(mode = LazyThreadSafetyMode.SYNCHRONIZED){
            DbHelper()
        }
    }

    init {
        val openHelper = DbOpenHelper(AppContext.getInstance(), DB_NAME, null)
        //获取数据库
        mDb = openHelper.writableDatabase
        //封装数据库中表的创建、更新、删除
        mDaoMaster = DaoMaster(mDb)  //合起来就是对数据库的操作
        //对表操作的对象。
        mSession = mDaoMaster.newSession() //可以认为是对数据的操作
    }

    fun getNewSession(): DaoSession {
        return mDaoMaster.newSession()
    }

}